File: 4214-count-dominant-indices/count-dominant-indices.py
class Solution:
    def dominantIndices(self, nums: List[int]) -> int:
        ans = 0
        count = 1
        RangeSum = nums[-1]

        for i in range(len(nums)-2 ,-1, -1):
            avg = RangeSum / count
            if ( avg < nums[i]):
                ans += 1
            count += 1
            RangeSum += nums[i]
        
        return ans
        

File: 4214-count-dominant-indices/count-dominant-indices.cpp
class Solution {
public:
    int dominantIndices(vector<int>& nums) 
    {
        int n = nums.size();

        int ans = 0;
        int rangeSum = nums[n-1];
        int count = 1;

        for(int i=n-2 ; i >= 0 ;--i)
        {
            int avg = rangeSum / count;
            if (nums[i] > avg) ans++;
            count++;
            rangeSum += nums[i];
        }
        return ans;

        
    }
};